### PR TITLE
Avatar Hash and Fingerprint

### PR DESCRIPTION
**Pull request by tomasz-ludek**
## Issue being fixed or feature implemented

NMA-698 v17 | Avatar Hash
NMA-697 v17 | Avatar Fingerprint
## Related PR's and Dependencies

https://github.com/dashevo/dash-wallet/pull/578
https://github.com/dashevo/android-dashpay/pull/35
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/ui/dashpay/work/UpdateProfileWorker.kt [hunk 1]
```diff
@@ -72,20 +72,27 @@ class UpdateProfileWorker(context: Context, parameters: WorkerParameters)
...
                     val fileId = saveToGoogleDrive(applicationContext, avatarFileBytes)
                     avatarUrl = "https://drive.google.com/uc?export=view&id=$fileId"
                 }
                 EditProfileViewModel.ProfilePictureStorageService.IMGUR.name -> {
                     //TODO:
                 }
             }
+        } else {
+            avatarHash = inputData.getByteArray(KEY_AVATAR_HASH)
         }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/work/UpdateProfileWorker.kt:96]: this is leftover code from a previous story.  It is not currently used.

File: wallet/src/de/schildbach/wallet/ui/dashpay/utils/DHash.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        if (resizedGrayscaleBmp != resizedBmp) {
+            resizedBmp.recycle();
+        }
+
+//        String fileName = "test1.png";
+//        File file = new File(storageDir, fileName);
+//        resizedGrayscaleBmp = BitmapFactory.decodeFile(file.getPath());
+
+        String dHashH = getHorizontalDifferences(resizedGrayscaleBmp);
+//        String dHashV = getVerticalDifferences(resizedGrayscaleBmp);
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/utils/DHash.java:51]: Some commented code here.

Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/dashpay/utils/DHash.java]: Fixed

File: wallet/src/de/schildbach/wallet/ui/dashpay/work/UpdateProfileWorker.kt (hunk 1 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/dashpay/work/UpdateProfileWorker.kt:96]: What do you mean by that @HashEngineering?
Are you referring to a particular line or rather to the whole code?

Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/dashpay/work/UpdateProfileWorker.kt:96]: The entire block of code.  We are not currently performing the image upload with the `UpdateProfileWorker`.  As can be seen in this block, the `IMGUR` section has no code.

This line can be put outside this block:
`            avatarHash = inputData.getByteArray(KEY_AVATAR_HASH)`

Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/dashpay/work/UpdateProfileWorker.kt]: Fixed

**Pull request merged**